types/merge_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) OpenFaaS Author(s) 2019. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package types
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_mergeDerekRepoConfigs_OnlyOneItem(t *testing.T) {
+
+	config1 := DerekRepoConfig{
+		ContributingURL: "http://example.com",
+	}
+
+	config2 := DerekRepoConfig{
+		ContributingURL: "http://two.example.com",
+	}
+
+	configOut, err := MergeDerekRepoConfigs(config1, config2)
+
+	if err != nil {
+		t.Errorf("Got error for a single plan, expected no error: %s", err.Error())
+		t.Fail()
+	}
+
+	if config2.ContributingURL != configOut.ContributingURL {
+		t.Errorf("ContributingURL want: %s, but got: %s", config2.ContributingURL, configOut.ContributingURL)
+	}
+}
+
+func Test_mergeDerekRepoConfigs_MergeEmptyItemsFromBoth(t *testing.T) {
```

Comment:
What happens if the 2nd repo has no ContributingURL ? Can you show that the base ContributingURL is still honoured?

Also check all structs for `omitempty` please

Reply:
Is that not what the config1.Redirect is showing? It doesnt exist in the second so it's not overridden.